histogram.go

```diff
@@ -0,0 +1,42 @@
+package imaging
+
+import (
+	"image"
+)
+
+// Histogram returns a normalized histogram of an image.
+//
+// Resulting histogram is represented as an array of 256 floats, where
+// histogram[i] is a probability of a pixel being of a particular luminance i.
+func Histogram(img image.Image) [256]float32 {
+	src := toNRGBA(img)
+	width := src.Bounds().Max.X
+	height := src.Bounds().Max.Y
+
+	var histogram [256]float32
```

Comment:
Please check the code for data races (e.g. `go test -race`). I think we shouldn't use `parallel` if we mutate this array inside the loops.